### PR TITLE
Chore: Add .golangci.yaml file

### PR DESCRIPTION
**Pull request by tanryberdi**
- added .golangci.yaml file
Signed-off-by: Tanryberdi <tanryberdi@gmail.com>

See #4

### CODE REVIEW

**Comment by tanryberdi** [comment]
@vaguecoder could you please review? 🙏  I disabled many linters, for examples (lll - length of line), we did not decide length of line, wdyt?

**Comment by vaguecoder** [comment]
@tanryberdi A few things to update here. I'll update on respective files shortly.

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,7 @@
+lint:
+	@echo "Linting ..."
```
Comment (vaguecoder) [inline Makefile:2]: The linter `loggercheck` as per [this][1] was available [golangci-lint v1.49.0][2]. However, I see it in effect only in the next version [golangci-lint v1.50.0][3] (which happens to be a major). Else, it fails with the following error:
```
Linting ...
ERRO Running error: unknown linters: 'loggercheck', run 'golangci-lint help linters' to see the list of supported linters 
make: *** [Makefile:3: lint] Error 3
```

I suggest to add a line in `make lint` something similar to: 
```
Please make sure you are on golangci-lint v1.50.0 or later:\n\tgo install github.com/golangci/golangci-lint/cmd/golangci-lint@latest\n`
```

[1]: https://golangci-lint.run/usage/linters/#disabled-by-default
[2]: https://github.com/golangci/golangci-lint/releases/tag/v1.49.0
[3]: https://github.com/golangci/golangci-lint/releases/tag/v1.50.0

**Comment by vaguecoder** [comment]
The comments from `.golangci.yml` are overloaded, even for an open-source project. Please remove all comments including the licensing part above.

The only things, as per me, require comments here are:
1. `issues` section. Not the default values and all, but if you have excluded some linter for a specific reason.
2. `linters-settings` section. Not the default values and all, but if you have linter configured.

I just did some trial and error and found the following are the minimum settings we require as of now for the above 2 sections:
```
linters-settings:
  cyclop:
    max-complexity: 30
  nolintlint:
    allow-unused: true

issues:
  exclude-rules:
    - path: "_test\\.go"
      linters:
        - bodyclose
        - noctx
```

I'm turning my back on my [original request][1]. Let's stick to this minimal settings and we can have future issues to add or configure more linters.

Please **keep the linters** that you've enabled so far and remove the comments describing them.

[1]: https://github.com/vaguecoder/gorilla-mux/issues/4#issuecomment-1374890213

**Comment by vaguecoder** [comment]
Also, I ask one more request of you on the same. Please add `TODO` against
1. `cyclop` - to reduce the complexity
2. `nolintlint` - to fix this (I believe you added inline comments wherever `nolint` is specified)
3. `bodyclose` - to close the body
4. `noctx` - to check if we can really use `context` in these places

We'll address these in future issues.

File: Makefile (hunk 1 above)
Comment (tanryberdi) [reply Makefile:2]: where I must to add this line? btw, it is last version go `golangci-lint v1.50.0`

Comment (vaguecoder) [reply Makefile:2]: Add the line to print in the same target, `make lint`.
```
lint:
         @echo "Linting ...Please make sure you are on golangci-lint v1.50.0 or later:\n\tgo install github.com/golangci/golangci-lint/cmd/golangci-lint@latest\n`"
```
In [golangci-lint v1.50.0](https://github.com/golangci/golangci-lint/releases/tag/v1.50.0) and [latest (golangci-lint v1.50.1)](https://github.com/golangci/golangci-lint/releases/tag/v1.50.1). The issue occurs when using older versions.

Comment (tanryberdi) [reply Makefile]: @vaguecoder done

Comment (vaguecoder) [reply Makefile]: Please also check the 2 comments below

Comment (tanryberdi) [reply Makefile]: done

**Review by vaguecoder — APPROVED** [review]
LGTM

**Pull request merged**